### PR TITLE
Adopt license eye to check dependencies' license

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/changelog/docs/en/changes/changes.md).

This is still not perfect, for example there are some duplicate dependencies that the maven resolver needs more enhancements, but that doesn't matter too much IMO.

### CODE REVIEW

File: docs/en/guides/README.md [hunk 1]
```diff
@@ -119,11 +119,12 @@ You would most likely have to add a new source and scope. To learn how to do thi
...
-- Check the LICENSE and NOTICE of those dependencies to make sure that they can be included in an ASF project. Add them to the `apm-dist/release-docs/{LICENSE,NOTICE}` file.
-- Add the names of these dependencies to the `tools/dependencies/known-oap-backend-dependencies.txt` file (**in alphabetical order**). `check-LICENSE.sh` should pass in the next run.
+We use [license-eye](https://github.com/apache/skywalking-eyes) to help you make sure that you haven't missed out any new dependencies:
+- Install `license-eye` according to [the doc](https://github.com/apache/skywalking-eyes#usage).
+- Run `license-eye dependency resolve --summary ./dist-material/release-docs/LICENSE.tpl` in the root directory of this project.
+- Check the modified lines in `./dist-material/release-docs/LICENSE` (via command `git diff -U0 ./dist-material/release-docs/LICENSE`) and
+  check whether the new dependencies' licenses are compatible with Apache 2.0.
+- Add the new dependencies' notice (if any) to `./dist-material/release-docs/NOTICE`, copy their license file to `./dist-material/release-docs/licenses` if they are not standard Apache 2.0 license.
```
Comment (wu-sheng) [inline docs/en/guides/README.md:127]:
```suggestion
- Add the new dependencies' notice files (if any) to `./dist-material/release-docs/NOTICE` if they are Apache 2.0 license. Copy their license files to `./dist-material/release-docs/licenses` if they are not standard Apache 2.0 license.
```

Comment (wu-sheng) [reply docs/en/guides/README.md:127]: NOTICE file only matters for Apache 2.0

Comment (kezhenxu94) [reply docs/en/guides/README.md:127]: > Suggested change 
> - Add the new dependencies' notice (if any) to `./dist-material/release-docs/NOTICE`, copy their license file to `./dist-material/release-docs/licenses` if they are not standard Apache 2.0 license.
> - Add the new dependencies' notice files (if any) to `./dist-material/release-docs/NOTICE` if they are Apache 2.0 license. Copy their license files to `./dist-material/release-docs/licenses` if they are not standard Apache 2.0 license.

> Add the new dependencies' notice files (if any) to `./dist-material/release-docs/NOTICE` if they are Apache 2.0 license.

Do you think this is correct? We only need to copy the notice if they are apache 2.0 license?

Comment (wu-sheng) [reply docs/en/guides/README.md:127]: Yes, I am sure. Notice file is mentioned in Apache 2.0 License. And most other licenses are not mentioning this. At least most widely used licenses are not, such as MIT, BSD 2/3

Comment (wu-sheng) [reply docs/en/guides/README.md]: No, we should not.

Let's take a look at the full content of MIT, https://opensource.org/licenses/MIT.
The whole text, it doesn't mention anything about the NOTICE file. 

Then let's see the APLv2, https://github.com/apache/skywalking/blob/0ac9874e79e1826e352f2c4a05c93aed7234700e/LICENSE#L106
> (d) If the Work includes a "NOTICE" text file as part of its distribution, ...

APLv2 declares the NOTICE file is actually a part of the LICENSE file, which is why we need to copy.

Comment (wu-sheng) [reply docs/en/guides/README.md]: I remember there was a discussion between Justin and me when DolphinScheduler was still incubating. But I can't find the thread anymore. But the contents of the license should be official guidance, and I can remember this principle due to this.

File: dist-material/release-docs/LICENSE.tpl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+========================================================================
+The following files are provided under the Apache-2.0 License. See project link for details.
+The text of each license is the standard Apache 2.0 license.
+
+    proto files from cncf/udpa: https://github.com/cncf/udpa Apache 2.0
+    proto files from envoyproxy/data-plane-api: https://github.com/envoyproxy/data-plane-api  Apache 2.0
+    proto files from prometheus/client_model: https://github.com/prometheus/client_model Apache 2.0
+    flatbuffers files from istio/proxy: https://github.com/istio/proxy Apache 2.0
+    svg files from skywalking-ui/src/assets/icons: https://github.com/google/material-design-icons Apache 2.0
+    file ZipkinQueryHandler.java reference from zipkin2.server.internal.ZipkinQueryApiV2: https://github.com/openzipkin/zipkin  Apache 2.0
```
Comment (wu-sheng) [inline dist-material/release-docs/LICENSE.tpl:47]: Why these are added in the binary license?

File: dist-material/release-docs/LICENSE.tpl [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+The Apache SkyWalking project contains subcomponents with separate copyright
+notices and license terms. Your use of the source code for the these
+subcomponents is subject to the terms and conditions of the following
+licenses.
+========================================================================
+
+{{ range .Groups }}
+========================================================================
+{{ .LicenseID }} licenses
```
Comment (wu-sheng) [inline dist-material/release-docs/LICENSE.tpl:14]: One question, what is the sequence of this license and repo name? Alphabeta?

**Review by wu-sheng — APPROVED** [review]

File: dist-material/release-docs/LICENSE.tpl (hunk 3 above)
Comment (kezhenxu94) [reply dist-material/release-docs/LICENSE.tpl:14]: They are ordered by dependency name.

File: .github/workflows/skywalking.yaml [hunk 4]
```diff
@@ -52,30 +52,62 @@ jobs:
         with:
           submodules: true
       - name: Check code style
-        run: ./mvnw clean checkstyle:check
+        run: ./mvnw -B -q clean checkstyle:check
+
+  dependency-license:
+    if: |
+      always() &&
+      ((github.event_name == 'schedule' && github.repository == 'apache/skywalking') || needs.changes.outputs.pom == 'true')
```
Comment (wu-sheng) [inline .github/workflows/skywalking.yaml:60]: Does this skipping make this PR not changed?

Comment (kezhenxu94) [reply .github/workflows/skywalking.yaml:60]: > Does this skipping make this PR not changed?

I don’t understand what you mean 😄

Comment (wu-sheng) [reply .github/workflows/skywalking.yaml:60]: GHA shows dependencies check skipped.

**Comment by wu-sheng** [comment]
B87A0F06-8E4C-4FC2-BBFD-77EBDCDB3DD0

**Comment by kezhenxu94** [comment]
> B87A0F06-8E4C-4FC2-BBFD-77EBDCDB3DD0

this is expected, we run the dependency check only when there is any `pom.xml` file changed, which is not the case in this PR

**Comment by kezhenxu94** [comment]
I'll very soon open another PR with `pom.xml` file changed to verify this.

**Pull request merged**